Jint/Engine.cs

```diff
@@ -382,6 +534,28 @@ public JsValue GetCompletionValue()
             return _completionValue;
         }
 
+        /// <summary>
+        /// Gets the last evaluated statement completion value, if it is a Promise unwraps it
+        /// </summary>
+        public JsValue GetPromiseCompletionValue()
```

Comment:
Why isn't it part of `GetCompletionValue()` directly? (unless I find the reason in one of your older comments)

Reply:
Then it will seem "funny" relative to other places.
e.g.
```csharp
engine.Execute("Promise.resolve(1);").GetCompletionValue()
```
Will return a `PromiseInstance` or `1`?

So I decided that it makes more sense to have it as explicit unwrapping. It is possible that it should only work with Promises. like so

```csharp
 public JsValue UnwrapPromise(PromiseInstance promise)
 ```

What do you think?

Reply:
In this case the promise, and if you do "await Promise.resolve(1)" then it returns the value.

Reply:
`await Promise.resolve(1)` is semantically equivalent to just `Promise.resolve(1)`. So I would personally expect them to behave the same way

Reply:
Try it in the chrome dev tools (or Safari?), and you'll see the difference there too. Even MDN says so: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/resolve#return_value
and so does the spec: https://262.ecma-international.org/10.0/#sec-promise-resolve

Can't beat that, right?

Reply:
Just to say that it won't be a surprise if the Promise is not unwrapped automatically. Only when `await` is used on it.

Reply:
My understand is `await Promise.resolve(1)` is using top-level await feature. Without it `await Promise.resolve(1)` is not even valid syntax, at least that is my current understanding

Reply:
settled with "public JsValue UnwrapIfPromise(this JsValue value)" extension method on `JsValue`